### PR TITLE
feat: Initial implementation of Gladia transcriptions to BBB 2.7

### PR DESCRIPTION
**Pull request by lfzawacki**
Still a work in progress, adds few UI/UX elements to support gladia options and a new message used for configuring the transcription backend.

Should be used with this branch of [bbb-transcriptions-controller](https://github.com/bigbluebutton/bbb-transcription-controller/tree/gladia)

HTML5 settings in setting.yml:

To enable Gladia support in the HTML5 client:

     audioCaptions:
       enabled: true
       # makes the CC button visible by default
       alwaysVisibile: true
       provider: gladia

For the transcription pad:

     captions:
       enabled: true
       # Hides the old CC button that was used to open the transcription pad.
       # This makes less sense when used together with transcriptions so we hide it
       showButton: false 

The current transcription settings can be configured via userdata parameters on join. Here are the parameters:

    userdata-bbb_transcription_partial_utterances -> Boolean: [true, false]
        Enables/disables the use of partial utterance results. These
        are lower quality results that have the advantage of arriving
        faster than the final, and more accurate, results.
        
        The default value is *true*.

    userdata-bbb_transcription_min_utterance_length -> Integer: [0,5]
        This value is measured in seconds and denotes the minimum
        size of a partial utterance for it to be considered as valid. Lower
        values will yield quicker and more inacurrate results and vice-versa.
        This can be a good way to balance speed with more accurate results.

        The default value is *3*

    userdata-bbb_transcription_provider -> [webspeech, gladia, vosk]
        Sets the transcription provider for this user. 'webspeech' will use
        the old browser based system which is enabled by default on
        most BBB instances. Both 'gladia' and 'vosk' will make BBB use
        the new 'transcription controller' system. 

## TODO

* Long results coming from gladia should be split so we don't lose any text

### CODE REVIEW

File: bigbluebutton-html5/imports/ui/components/user-list/user-list-content/user-participants/user-options/component.jsx [hunk 1]
```diff
@@ -328,6 +328,7 @@ class UserOptions extends PureComponent {
           dataTest: 'writeClosedCaptions',
         });
       }
+
```
Comment (antobinary) [inline bigbluebutton-html5/imports/ui/components/user-list/user-list-content/user-participants/user-options/component.jsx:331]:
```suggestion
```

File: bigbluebutton-html5/private/config/settings.yml [hunk 2]
```diff
@@ -238,6 +239,9 @@ public:
       dataSaving:
         viewParticipantsWebcams: true
         viewScreenshare: true
+      transcription:
+        partialUtterances: true
```
Comment (antobinary) [inline bigbluebutton-html5/private/config/settings.yml:243]: Please provide a 1 line comment above explaining what partial utterances are -- for administrators' sake

File: bigbluebutton-html5/imports/ui/components/captions/container.jsx [hunk 3]
```diff
@@ -11,20 +11,6 @@ const Container = (props) => {
   const { isResizing } = cameraDock;
   const layoutContextDispatch = layoutDispatch();
 
-  const { amIModerator } = props;
```
[the remark this answers is not in the document]
Comment (lfzawacki) [reply bigbluebutton-html5/imports/ui/components/captions/container.jsx:14]: The code that used this was removed. What it originally did was close the transcription pad for non moderators, but we want it to be visible.

File: bigbluebutton-html5/imports/api/users-settings/server/methods/addUserSettings.js [hunk 4]
```diff
@@ -65,6 +65,9 @@ const currentParameters = [
   'bbb_hide_nav_bar',
   'bbb_change_layout',
   'bbb_direct_leave_button',
+  // TRANSCRIPTION
+  'bbb_transcription_partial_utterances',
```
[the remark this answers is not in the document]
Comment (lfzawacki) [reply bigbluebutton-html5/imports/api/users-settings/server/methods/addUserSettings.js:69]: Added some docs to the PR text

**Comment by jqueguiner** [comment]
Long results can be handled with endpointing. We are thinking also to add an option to force transcription to maximum Xsec. It should take the min(speech_with_endpointing_trigger, max_Xsec)

**Pull request merged**